### ISSUE TITLE
vue-mode frequently loses the plot with respect to indentation

### ISSUE THREAD

**Issue by glyph** [AdamNiederer/vue-mode#70]
Especially on larger component files ("more than 200 lines") I seem to be running into situations where vue-mode just stops indenting (no error, no message in `*Messages*`); hitting <kbd>Tab</kbd> just puts the current line on the left margin.  `M-x normal-mode` fixes this, but there's a second or two hit as everything gets reparsed.

Perhaps unrelatedly, `indent-region` seems to do something weird in `<template>` sections. The coding standard I work with expects

```
<template>
    <div>
        <span>some content</span>
    </div>
</template>
```

but vue-mode *really* wants this to instead be

```
<template>
<div>
    <span>some content</span>
</div>
</template>
```

and will reformat my *entire* template section to its preferred style if I ever use `indent-region`, even on 2 lines, rather than `indent-for-tab-command`, which indents an individual line correctly.

I think the latter may be a configuration issue that I don't understand rather than a bug, but they are both part of the same bad experience with indentation for me at the moment :).

Otherwise, thanks a bunch for providing vue-mode, it was great to find this and just get up and running when I started to need to edit Vue stuff at work!

**Comment by AdamNiederer** [comment]
> Especially on larger component files ("more than 200 lines") I seem to be running into situations where vue-mode just stops indenting (no error, no message in *Messages*); hitting Tab just puts the current line on the left margin. M-x normal-mode fixes this, but there's a second or two hit as everything gets reparsed.

I'll look into this - I don't think I've ever tested it on anything larger than 300 or so lines, actually.

> Perhaps unrelatedly, indent-region seems to do something weird in <template> sections. 

The indentation is by design, but I've been meaning to add a setting which lets you indent the template section your way as well.

The idea is that for a section like this

```html
<template>
  <p>
  </p>
</template>
```

Yanking that section into a buffer (like with `vue-mode-edit-indirect-at-point`) causes the whole thing to be "indented wrong"

```html
  <p>
  </p>
```

Anyway, I'll hack together a config option in `vue-html-mode` to make this happen.

**Comment by AdamNiederer** [comment]
Should be resolved via 30c3650a31057a169dae74e10e394760a6119729 and dcee704e939a9ced2205370c2a7b032a2a69aa04. Once melpa rebuilds, customize `vue-html-extra-indent` to the number of spaces you want your entire template to be indented by (I think you want 4), and give it a shot.

**Comment by glyph** [comment]
I've customized that variable and it indeed seems to fix the `indent-region` in sections issue.  Thanks for the fix!  I haven't been doing as much frontend stuff as previously, so I'm not 100% sure that the >200-line-indent-breaking issue is totally fixed, but if it's related, it should be.

Since I inappropriately filed 2 separate issues here rather than splitting them out, I'm going to mark this as closed, and if I still have the "loses the plot" issue with the current mode, I'll file a new one.

Thanks again for all your work on this mode, it's made my work life much more pleasant!
